widget.py

```diff
@@ -0,0 +1,309 @@
+        super(My_slider, self).on_changed(self.on_check)  # On click reaction.
+
+    def on_check(self, event):
+        self.vmax = event
+        self.image.set_clim(vmax=self.vmax)
+        self.fig.canvas.draw()
+
+    def set_image(self, image):
+        self.image = image
+
+    def get_vmax(self):
+        return self.vmax
+
+    def get_vmin(self):
+        return self.vmin
+
+
+class Bttn(Button):
+    """
+    Class for button changing crystal type.
+    fig - whole image figure called by canvas.draw().
+    list_color - list with def. colours.
+    histogram_list - list with objects of class histogram.
+    dict_color_histogram - dictionary with keys as centering types and value
+    is the colour name.
+    """
+    def __init__(self, fig, axs, label, list_color,
+                 histogram_list, dict_color_histogram):
+        # Initialize parent constructor.
+        super(Bttn, self).__init__(ax=axs, label=label, color=list_color[0])
+        self.fig = fig  # For calling canvas draw which refreshes the image.
+        # Loop for the colour list of a given button.
+        self.colors = itertools.cycle(list_color)
+        super(Bttn, self).on_clicked(self.on_check)  # Click reaction
+        self.histogram_list = histogram_list  # List with all histograms
+        self.dict_color_histogram = dict_color_histogram
+        self.list_color = list_color
+
+    def on_check(self, label):
+        """
+        Change colour of the histogram depending
+        on which switch has been clicked.
+        """
+        self.color = next(self.colors)  # After clicking changes the colour
+        # to the next in a loop. For a given button the colour will be set in
+        # the histogram.
+        # change name color in diction for key PABCFHIR is label button
+        self.dict_color_histogram[self.label.get_text()] = self.color
+        for hist in self.histogram_list:  # Loop for each histogram.
+            hist.set_colour(self.dict_color_histogram)  # Setting new colour
+            # in a given histogram.
+            # Refresh each histogram (clear image and draw again)
+            hist.update_color()
+        self.fig.canvas.draw()  # Redraw the current figure.
+
+    def reset_color(self):
+        self.colors = itertools.cycle(self.list_color)
+        self.color = self.list_color[0]
+        self.dict_color_histogram[self.label.get_text()] = self.color
+        for hist in self.histogram_list:  # Loop for each histogram.
+            hist.set_colour(self.dict_color_histogram)  # Setting new colour
+            # in a given histogram.
+            # Refresh each histogram (clear image and draw again)
+            hist.update_color()
+
+
+class Span():
+    """
+    Class for selecting the region of interest.
+    which_was_clicked_before - list of flags for showing
+        what was selected where.
+    dict_witch_range_crystal - original scope under as value (?).
```

Comment:
docstring does not seem to fit the class

Reply:
Addressed in: e73a1996e75cfceb523bf223bbad6228e6e19c81